### ISSUE TITLE
Typehint parsing seems to strip typehints from anonymous class

### ISSUE THREAD

**Issue by jschreuder** [phpspec/phpspec#813]
When running [this](https://github.com/jschreuder/SpotApi/blob/master/spec/Application/Request/HttpRequestParserRouterSpec.php#L64) spec, it errors with the following message:

```
PHP Fatal error:  Declaration of class@anonymous::parseHttpRequest($httpRequest, array $attributes): Spot\Api\Application\Request\Message\RequestInterface must be compatible with Spot\Api\Application\Request\HttpRequestParserInterface::parseHttpRequest(Psr\Http\Message\ServerRequestInterface $httpRequest, array $attributes): Spot\Api\Application\Request\Message\RequestInterface in E:\Sandbox\SpotApi\spec\Application\Request\HttpRequestParserRouterSpec.php on line 64
```

So it looks like the trick used to parse out the typehints from the file also strips the typehint from my anonymous class.

[edit] This is on 2.4.0, when I go back to 2.3.0 it runs fine.

**Comment by ciaranmcnulty** [comment]
Yes this is probably true and needs to be fixed. We hadn't considered anonymous classes.

Priority for PHP 7 support was 'existing test suites pass' so we didn't cover off many actual PHP 7 features.

Is there a reason for using an anonymous class rather than a double, by the way?

**Comment by sroze** [comment]
That actually makes sense as all the typehints are removed. The easiest way to fix this bug is to keep the typehints in the anonymous declarations, as it makes non-sense to strip them.

**Comment by jschreuder** [comment]
> Is there a reason for using an anonymous class rather than a double, by the way?

It was a quick & dirty workaround. Those classes work with a Pimple container and assigning a Prophet object to them caused type errors if I remember correctly.

**Comment by sroze** [comment]
The issue is fixed by #815 @ciaranmcnulty!

**Comment by jschreuder** [comment]
Awesome, thanks @sroze!
